network/book.go

```diff
@@ -219,7 +219,7 @@ func (s *SimpleBook) Sample() ([]string, error) {
 func (s *SimpleBook) slice(active bool) []*entry {
 	entries := make([]*entry, 0)
 	for _, e := range s.entries {
-		if !active && e.active {
+		if e.active != active {
```

Comment:
This is not correct.
We either want to include active addresses or not; however, we always want to include not active ones.
The original code is correct in this case.